packages/backend/src/domain/services/discord-auth.ts

```diff
@@ -0,0 +1,140 @@
+import type { Context } from "hono";
+import { injectable } from "inversify";
+
+export interface IDiscordAuthService {
+  getAuthUrl(c: Context): Promise<string>;
+  authorization(c: Context, code: string): Promise<AuthorizationResponse>;
+  refreshToken(
+    c: Context,
+    refreshToken: string
+  ): Promise<AuthorizationResponse>;
+  getUserResource(
+    c: Context,
+    accessToken: string
+  ): Promise<DiscordUserResource>;
+}
+
+@injectable()
+export class DiscordAuthService implements IDiscordAuthService {
+  constructor() {}
+
+  private discordApiBaseUrl = "https://discordapp.com/api";
```

Comment:
定数ぽいのでアッパースネークがいいかと！
```suggestion
  private DISCORD_API_BASE_URL = "https://discordapp.com/api";
```